packages/cw-orch-core/src/environment/queriers/mod.rs

```diff
@@ -0,0 +1,64 @@
+use cosmwasm_std::{Addr, BlockInfo};
+use serde::{de::DeserializeOwned, Serialize};
+
+use self::{bank::BankQuerier, env::EnvironmentQuerier, node::NodeQuerier, wasm::WasmQuerier};
+use crate::CwEnvError;
+use std::fmt::Debug;
+
+pub mod bank;
+pub mod env;
+pub mod node;
+pub mod wasm;
+
+/// This trait acts as the high-level trait bound for supported queries on a `CwEnv` environment.
+/// It also implements some high-level functionality to make it easy to access.
+pub trait QueryHandler: DefaultQueriers {
+    type Error: Into<CwEnvError> + Debug;
+
+    /// Wait for an amount of blocks.
+    fn wait_blocks(&self, amount: u64) -> Result<(), Self::Error>;
```

Comment:
I suggest that this function and the others implemented firectly in this trait are simply a wrapper over underlying functions. So we can implemented them inside the trait directly !

Reply:
Actually most of them are very env-specific.

Reply:
I think block_info is one that can be implemented generic and possibly the wasm query. 

But the Error type is a problem. The trait has its own err type that requires a `Into<CwEnvError>` trait, which the errors returned by the trait (Querier) doesn't have. 

Do we want to add it or?